user/src/com/google/gwt/user/client/ui/RootPanel.java

```diff
@@ -126,10 +130,17 @@ public static void detachNow(Widget widget) {
    * contained in another widget. This is to ensure that the DOM and Widget
    * hierarchies cannot get into an inconsistent state.
    * </p>
-   * 
```

Comment:
Alternative:

Remove the added paragraph and then:

`@deprecated` This method was introduced to prevent memory leaks in very old browsers which is not relevant anymore. Also `unload` events are not fired reliably on mobile browsers and are not compatible with the back/forward cache (bfcache) in modern browers. If your application has code in `Widget.onUnload()` methods that needs to be executed if the window has been closed then consider using `pagehide` event instead. Note that `pagehide` events are compatible with bfcache but are still not fired reliably on mobile devices. Also note that bfcache effectively makes a memory dump on `pagehide` and a memory restore on `pageshow`. If you blindly execute cleanup code during `pagehide` your widget might not work correctly anymore once restored from bfcache. You should check `pagehide.persisted` before executing cleanup code and possibly also use `pageshow` together with `pageshow.persisted` to execute initialization code.

Reply:
That's a bit heavy for a deprecated tag, how do you feel about an abbreviated version of that, plus a link to 9908 for more info/discussion?

Reply:
I am fine with either way. I just wanted to ensure that a reader also knows the pitfalls of using `pagehide` as a replacement.